src/components/content/injection.tsx

```diff
@@ -28,27 +28,47 @@ export function Injection({ href }: InjectionProps) {
   useEffect(() => {
     const origin = window.location.host
     const protocol = window.location.protocol
-    void fetch(
-      `${protocol}//${origin}/api/frontend${encodeURI(
-        href.startsWith('/') ? href : `/${href}`
-      )}`
-    )
+    const encodedHref = encodeURI(href.startsWith('/') ? href : `/${href}`)
+
+    try {
+      const cachedData = sessionStorage.getItem(encodedHref)
+      if (cachedData) {
+        dataToState(JSON.parse(cachedData))
```

Comment:
not returning here meanings that we are still doing a request in the background, that is fine if we ...

Reply:
just return here :)